check.js

```diff
@@ -21,6 +21,7 @@ updateNotifier = new updateNotifier.UpdateNotifier(options);
 
 	// Call process exit explicitly to terminate the child process,
 	// otherwise the child process will run forever, according to the Node.js docs
+	timer.unref();
```

Comment:
This makes no difference here. `process.exit()` makes the process exit regardless of running event listeners.

Reply:
http://docs.libuv.org/en/v1.x/guide/utilities.html?highlight=ref#event-loop-reference-count

Reply:
Basically, there is a ref in libuv to determine whether the uv loop should exit automatically, as long as there is a ref, it will not exit:

Reply:
I'm familiar with how the event loop works, however, you have not provided any comment or proof that counteracts my previous comment.

Reply:
![image](https://user-images.githubusercontent.com/15144040/97866905-c7fc0d00-1d47-11eb-8ad6-aeaf3f810bce.png)

Reply:
The phenomenon is that the process cannot exit, just add unref.

Reply:
use ncc packages

Reply:
[reference](https://www.bennadel.com/blog/3301-calling-timeout-unref-with-settimeout-does-not-appear-to-be-a-best-practice-in-node-js.htm)

>  So, immediately, the Node.js documentation warns us that we should take caution when using the timeout.unref() feature. Which means that we-developers better have a really good reason for using .unref() in our Circuit Breaker code (or other such inspired code). But, I'm struggling to think of a good reason. In fact, I can only thing of reasons that .unref() seems silly.
   For starters, the underlying Promise may still hold the process open. If you look back at the previous terminal output, you can see the 1-Second timeout; but, you can also see that the underlying API call still returns after 5-Seconds. As such, a long-running command will hold the process open longer than its sibling timeout.
  Of course, if the command is resolved prior to the 1-Second timeout, we clear the pending timer both for Resolution and Rejection outcomes. As such, the timer will never hold the process open longer that the command execution itself, whether it's long-running or short-lived.

Reply:
Implicit and explicit exits are different.